Animal_Sound/Program.cs

```diff
@@ -1,2 +1,60 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿using System;
+using System.Collections.Generic;
+
+interface IAnimal
+{
+    string Nume { get; set; }
+    string Sunet { get; set; }
+    void AfiseazaSunet();
+}
+
+class Animal : IAnimal
+{
+    public string Nume { get; set; }
+    public string Sunet { get; set; }
+
+    public Animal(string nume, string sunet)
+    {
+        Nume = nume;
+        Sunet = sunet;
+    }
+
+    public void AfiseazaSunet()
+    {
+        Console.WriteLine($"{Nume} face sunetul: {Sunet}");
+    }
+}
+
+class Caine : Animal
+{
+    public Caine() : base("Caine", "Ham Ham HAM ham!") { }
```

Comment:
public Caine() : base("Batoza", "VRUM VRUM!") { } // my PO forced me to do this change

unde e incapsularea? unde sunt principiile OOP.